### DISCUSSION TITLE
config.yml support Variables?

### DISCUSSION THREAD

**Discussion by IrishTR** [meisnate12/Plex-Meta-Manager#2109] [General]
Forgive me as I am not finding this topic through search, I am wanting to use Environment Variables inside the config.yml file pulled into the container either via a .env file or from docker secrets file would be preferred.  Looking to achieve this to not expose all my API keys and passwords via clear text in the config.yml file.  I have this option in other containers I run like Notifiarr, Homepage, etc.  I am including a sample of what my compose file would optimally look like (I commented out the .ENV option from host and showing the docker secrets preferred method.

**Compose section:**

`  plexmm:
    <<: *common-keys-apps
    image: linuxserver/kometa
    container_name: plexmm
    profiles: ["media", "all"]
    depends_on:
      - plex
    environment:
      <<: *default-tz-puid-pgid
      KOMETA_CONFIG: /config/config.yml #optional
      KOMETA_TIMES: 05:00 #optional
      KOMETA_RUN: "False" #optional
      KOMETA_TEST: "False" #optional
      KOMETA_NO_MISSING: "False" #optional
      KOMETA_FILE_PLEX_TOKEN: /run/secrets/plex_token
      KOMETA_FILE_TRAKT_CLIENTID: /run/secrets/trakt_clientid #$TRAKT_CLIENTID
      KOMETA_FILE_TRAKT_CLIENTSECRET: /run/secrets/trakt_clientsecret #$TRAKT_CLIENTSECRET
      KOMETA_FILE_TRAKT_PIN: /run/secrets/trakt_pin #$TRAKT_PIN
      KOMETA_FILE_GITHUB_TOKEN: /run/secrets/github_token #$GITHUB_TOKEN
      KOMETA_FILE_RADARR_API: /run/secrets/radarr_api #$RADARR_API
      KOMETA_FILE_SONARR_API: /run/secrets/sonarr_api #$SONARR_API
      KOMETA_FILE_MDBLIST_API: /run/secrets/mdblist_api #$MDBLIST_API
      KOMETA_FILE_OMDB_API: /run/secrets/omdb_api #$OMDB_API
      KOMETA_FILE_TMDB_API: /run/secrets/tmdb_api #$TMDB_API
      KOMETA_FILE_TAUTULLI_API: /run/secrets/tautulli_api #$TAUTULLI_API
      SERVER_IP: $SERVER_IP
      PLEX_PORT: $PLEX_PORT
    volumes:
      - $DOCKERDIR/plex-meta-manager/config:/config:rw
    secrets:
      - plex_token
      - trakt_clientid
      - trakt_clientsecret
      - trakt_pin
      - github_token
      - sonarr_api
      - radarr_api
      - mdblist_api
      - omdb_api
      - tmdb_api
      - tautulli_api`

**config.yml example**

`plex:                            # Can be individually specified per library as well; REQUIRED for the script to run
  url: http://{{KOMETA_VAR_SERVER_IP}}:{{KOMETA_VAR_PLEX_PORT}}
  token: {{KOMETA_FILE_PLEX_TOKEN}}
  timeout: 60
  clean_bundles: false
  empty_trash: false
  optimize: false
  db_cache: 4096
  verify_ssl:
github:
  token: {{KOMETA_FILE_GITHUB_TOKEN}}
tmdb:                            # REQUIRED for the script to run
  apikey: {{KOMETA_FILE_TMDB_API}}
  language: en
  cache_expiration: 60
  region: US
tautulli:                        # Can be individually specified per library as well
  url: http://tautulli:8181
  apikey: {{KOMETA_FILE_TAUTULLI_API}}
omdb:
  apikey: {{KOMETA_FILE_OMDB_API}}
  cache_expiration: 60
mdblist:
  apikey: {{KOMETA_FILE_MDBLIST_API}}
  cache_expiration: 60
notifiarr:
  apikey: {{KOMETA_FILE_NOTIFIARR_API}}
radarr:                          # Can be individually specified per library as well
  url: http://radarr:7878/radarr
  token: {{KOMETA_FILE_RADARR_API}}
  add_missing: false
  add_existing: false
  root_folder_path: /movies
  monitor: true
  availability: announced
  quality_profile: HD Bluray + WEB
  tag:
  search: true
  radarr_path: /movies
  plex_path: /media/movies
  upgrade_existing: false
  ignore_cache: false
  monitor_existing: false
radarr2:                          # Can be individually specified per library as well
  url: http://radarr:7878/radarr
  token: {{KOMETA_FILE_RADARR_API}}
  add_missing: false
  add_existing: false
  root_folder_path: /movies
  monitor: true
  availability: announced
  quality_profile: UHD Bluray + WEB
  tag:
  search: true
  radarr_path: /movies
  plex_path: /media/movies
  upgrade_existing: false
  ignore_cache: false
sonarr:                          # Can be individually specified per library as well
  url: http://sonarr:8989/sonarr
  token: {{KOMETA_FILE_SONARR_API}}
  add_missing: false
  add_existing: false
  root_folder_path: /tv
  monitor: all
  quality_profile: WEB-DL (1080p)
  language_profile: English
  series_type: standard
  season_folder: false
  tag:
  search: true
  cutoff_search: false
  sonarr_path: /tv
  plex_path: /media/tvshows
  upgrade_existing: false
  ignore_cache: false
  monitor_existing: false
trakt:
  client_id: {{KOMETA_FILE_TRAKT_CLIENTID}}
  client_secret: {{KOMETA_FILE_TRAKT_CLIENTSECRET}}
  pin: {{KOMETA_FILE_TRAKT_PIN}} 
  authorization:
    access_token: #########################
    token_type: Bearer
    expires_in: ########
    refresh_token: ##########
    scope: public
    created_at: #########`

**Comment by IrishTR** [comment]
GAH Apologies I tried using Code snippet but it looks like it still screwed up the formatting

**Comment by chazlarson** [comment]
https://kometa.wiki/en/nightly/kometa/environmental/#kometa-vars

config.yml
```
plex:
  url: http://<<PLEXSERVERIP>>:<<PLEXPORT>>
  token: <<PLEXTOKEN>>
  timeout: 60 
  clean_bundles: false 
  empty_trash: false 
  optimize: false 
  db_cache: 4096 
  verify_ssl: 
```

docker-compose:
```
  kometa-01:
    image: kometateam/kometa
...
    environment:
      KOMETA_PLEXSERVERIP: ${SOME_HOST_ENV_VAR_01}
      KOMETA_PLEXPORT: ${SOME_HOST_ENV_VAR_02}
      KOMETA_PLEXTOKEN: ${SOME_HOST_ENV_VAR_03}
```

**Reply by IrishTR** [reply]
Thanks can't believe I totally overlooked that link you provided.  Appreciate the examples I'll go give them a whirl!

**Reply by IrishTR** [reply]
Ok got it all converted over, only issue it did not like using  the PLEXPORT when running logs would error out saying it couldn't parse the URL.  Everything else working solid with .ENV insertion thanks again.
